packages/keyring/src/pair/index.ts

```diff
@@ -127,8 +125,9 @@ export function createPair ({ toSS58, type }: Setup, { publicKey, secretKey }: P
     get addressRaw (): Uint8Array {
       const raw = TYPE_ADDRESS[type](publicKey);
 
-      return type === 'ethereum'
-        ? raw.slice(-20)
+      return type === 'ethereum' &&
+          raw.length > 20
+        ? (raw).slice(-20)
```

Comment:
```suggestion
      return type === 'ethereum'
        ? raw.slice(-20)
```

Original fine as well, if 20 bytes still works exactly as expected.